presto-product-tests/src/main/java/com/facebook/presto/tests/SqlCancelTests.java

```diff
@@ -0,0 +1,186 @@
+        if (queryCanceller != null) {
+            queryCanceller.close();
+        }
+    }
+
+    private class SqlExecuteCallable
+            implements Callable<Void>
+    {
+        private String sql;
+
+        public SqlExecuteCallable(String sql)
+        {
+            this.sql = sql;
+        }
+
+        public Void call()
+        {
+            assertThat(() -> query(sql)).failsWithMessage("Query was canceled");
+            return null;
+        }
+    }
+
+    private class SqlCancelCallable
+            implements Callable<Response>
+    {
+        private String sqlStatement;
+
+        public SqlCancelCallable(String sqlStatement)
+        {
+            this.sqlStatement = sqlStatement;
+        }
+
+        public Response call()
+        {
+            long endTime = System.currentTimeMillis() + WAIT_TIME;
+            while (System.currentTimeMillis() < endTime && !Thread.currentThread().isInterrupted()) {
+                QueryResult queryResult = query(format("SELECT state, query_id from system.runtime.queries" +
+                        " WHERE query = '%s' ORDER BY created DESC LIMIT 1", sqlStatement));
+                if (queryResult.getRowsCount() == 0) {
+                    sleepUninterruptibly(STEP, TimeUnit.MILLISECONDS);
+                    continue;
+                }
+                String queryState = (String) queryResult.row(0).get(0);
+                String queryId = (String) queryResult.row(0).get(1);
+                if (!queryState.equalsIgnoreCase("running")) {
+                    sleepUninterruptibly(STEP, TimeUnit.MILLISECONDS);
+                }
+                else {
+                    return queryCanceller.cancel(queryId);
+                }
+            }
+            return null;
+        }
+    }
+
+    private void doCancelTest(String sql)
+            throws Exception
+    {
+        Future<Void> executeQueryThread = executor.submit(new SqlExecuteCallable(sql));
+        Future<Response> cancelQueryThread = executor.submit(new SqlCancelCallable(sql));
+
+        try {
+            executeQueryThread.get(2, TimeUnit.MINUTES);
+        }
+        catch (TimeoutException e) {
+            executeQueryThread.cancel(true);
+            throw new RuntimeException("Query execution time limit exceeded", e);
+        }
+
+        Response response = cancelQueryThread.get();
+        assertNotNull(response, "Error in cancel query thread");
+
+        if (response.getStatusCode() != HttpStatus.OK.code() && response.getStatusCode() != HttpStatus.ACCEPTED.code() &&
+                response.getStatusCode() != HttpStatus.NO_CONTENT.code()) {
+            fail("Unexpected error code " + response.getStatusCode() + "; reason=" + response.getStatusMessage());
+        }
+    }
+
+    @Test(groups = {CANCEL_QUERY})
+    public void cancelCreateTableTest()
+            throws Exception
+    {
+        String tableName = format("cancel_createtable_%s", randomUUID().toString().replace("-", ""));
```

Comment:
using random integer (or `nanoTime()` as we do elsewhere) would be simpler, no need for `.toString().replace("-", "")` then